pubtools/_pulp/tasks/delete.py

```diff
@@ -0,0 +1,709 @@
+            )
+            # TODO: collect units "SKIPPING"
+
+        return repo_map
+
+    def log_items(self, repo_map):
+        item_map = {}
+
+        for repo, items in repo_map.items():
+            for item in items:
+                item_map.setdefault(item, []).append(repo)
+
+        for item, repos in sorted(item_map.items()):
+            LOG.info("Deleting %s from %s", item.name, repos)
+
+        return repo_map
+
+    @step("Process push items")
+    def process_pushitems(self, pushitems, repos):
+        repo_map = {}
+
+        for item in pushitems:
+            if not isinstance(item, RpmPushItem):
+                LOG.warning("%s is not and RpmPushItem. Skipped for removal", item.name)
+                continue
+            mapped = False
+            for repo in repos or item.dest:
+                if repo in item.dest:
+                    repo_map.setdefault(repo, []).append(item)
+                    mapped = True
+
+            if not mapped:
+                LOG.warning(
+                    "%s is not present in any repo: %s", item.name, ", ".join(repos)
+                )
+                item = attr.evolve(item, state="SKIPPED")
+
+        return repo_map
+
+    @step("Verify repos from advisory")
+    def verify_repos_from_advisory(self, advisory, repo_map):
+        repo_membership = advisory.repository_memberships
+
+        for repo_name in list(repo_map.keys()):
+            if repo_name not in repo_membership:
+                LOG.warning(
+                    "Advisory %s is not in repo %s. Skipping the corresponding repo contents for removal",
+                    advisory.id,
+                    repo_name,
+                )
+                repo_map.pop(repo_name)
+
+        return repo_map
+
+    @step("Remove artifacts from module")
+    def remove_mod_artifacts(self, unit_map, signing_keys):
+        out = []
+        for item in unit_map.values():
+            rpms = item.unit.artifacts_filenames
+            repos = item.repos
+            if rpms:
+                out.append(self.delete_rpms(rpms, signing_keys, repos))
+
+        return out
+
+    @step("Get advisory from Pulp")
+    def get_advisory(self, advisory_id):
+        criteria = criteria = Criteria.and_(
+            Criteria.with_unit_type(ErratumUnit), Criteria.with_id(advisory_id)
+        )
+        adv_f = self.search_content(criteria)
+        advisory = [a for a in adv_f.result()]
+        if not advisory:
+            self.fail("Advisory %s not found", advisory_id)
+
+        return advisory[0]
+
+    @step("Get RPMs")
+    def get_rpms(self, rpm_names, signing_keys):
+        criteria = self.unit_criteria(
+            RpmUnit, self._rpm_search_criteria(rpm_names, signing_keys)
+        )
+        rpms_f = self.search_content(criteria)
+        rpms_f = f_map(
+            rpms_f,
+            partial(self.log_missing_units, unit_names=rpm_names, unit_attr="filename"),
+        )
+        return rpms_f
+
+    @step("Get ISOs")
```

Comment:
Could you please call this "files" instead of "ISOs" ?  We should try to move away from the "ISO == generic file" terminology which doesn't make sense.

Reply:
Sure.

Reply:
Done